packages/build/tests/mutate_save/tests.js

```diff
@@ -196,6 +196,60 @@ test('--saveConfig saves the configuration changes as netlify.toml', async (t) =
   }
 })
 
+test('--saveConfig does not truncate high amount of redirects', async (t) => {
+  const fixtureDir = `${FIXTURES_DIR}/many_redirects`
+  const fixtureConfigPath = `${fixtureDir}/netlify.toml`
+  const configPath = `${fixtureDir}/test_netlify.toml`
+  await fs.copyFile(fixtureConfigPath, configPath)
+  const { address, stopServer } = await startDeployServer({
+    async onRequest() {
+      const newConfigContents = await fs.readFile(configPath, 'utf8')
+      t.true(newConfigContents.includes('999'))
```

Comment:
The test fixtures has 1000 redirects/headers. If `999` is present in `netlify.toml`, it should indicate that all redirects/headers have not been truncated.